### PR TITLE
CE-1425 Track first page views of power users

### PR DESCRIPTION
**Pull request by adamkarminski**
[See CE-1425](https://wikia-inc.atlassian.net/browse/CE-1425)

We need to know more about our Power Users. One of the questions we need an answer to is: "Where the heck do they even start?!" Do they use Special:RecentChanges? Do they use Special:WikiActivity? Or maybe jump right into favorite articles? Let's find out!

The code works like this:
- If a PowerUser visits us - there is a JS tracking asset added to the output
- If there is no cookie that counts their page views - it sets one up with a value '1'
- If the cookie's value is lower than a maximum count (5 at the moment) - it launches an internal tracking mechanism
- The cookie exists for `time() + 3 hours` and is renewed with every page view. However, if a user is not active for _3h+_ it will treat his visit as a 'new session' and set up the cookie and launch the tracking.

@Wikia/community-engineering @macbre

### CODE REVIEW

File: includes/wikia/traits/PowerUserTrait.php [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 trait PowerUserTrait {
 	// Required base-class methods
-	abstract function getBoolOption();
+	abstract function getBoolOption( $oname );
```
Comment (gabrys) [inline includes/wikia/traits/PowerUserTrait.php]: sName + docs?

File: extensions/wikia/PowerUser/js/PowerUserPageViewsTracking.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+			cookieExpires = 1000 * 60 * 60 * 3;
+
+		function init() {
+			if (!pvIsCookieSet()) {
+				pvSetCookieCounter(cookieDefaultValue);
+			}
+
+			if (pvGetCookieCounter() <= maxCounter) {
+				pvLaunchTracking();
+				pvIncreaseCookieCounter();
```
Comment (gabrys) [inline extensions/wikia/PowerUser/js/PowerUserPageViewsTracking.js]: increase always?

Comment (adamkarminski) [inline extensions/wikia/PowerUser/js/PowerUserPageViewsTracking.js]: D'oh! True.

File: includes/wikia/traits/PowerUserTrait.php (hunk 1 above)
Comment (macbre) [inline includes/wikia/traits/PowerUserTrait.php]: `public function getBoolOption( $oname )` is how this method is defined in `User` class. Let's keep consistent :)

**Comment by adamkarminski** [comment]
unitTests_js_branches have been (rerun with a success)[http://qa-s2.wikia-prod:8080/job/unitTests_js_branches/10062/testReport/]

**Comment by gabrys** [comment]
LGTM

**Pull request merged**